pkg/test/framework/components/ingress/kube.go

```diff
@@ -42,65 +45,99 @@ var (
+	// port.
+	n := cfg.Istio.Settings().IngressNamespace
+	if env.Settings().Minikube {
+		pods, err := env.GetPods(n, fmt.Sprintf("istio=%s", istioLabel))
+		if err != nil {
+			return nil, false, err
+		}
 
-	env := ctx.Environment().(*kube.Environment)
+		scopes.Framework.Debugf("Querying ingress, pods:\n%v\n", pods)
+		if len(pods) == 0 {
+			return nil, false, errors.New("no ingress pod found")
+		}
 
-	address, err := retry.Do(func() (interface{}, bool, error) {
+		scopes.Framework.Debugf("Found pod: \n%v\n", pods[0])
+		ip := pods[0].Status.HostIP
+		if ip == "" {
+			return nil, false, errors.New("no Host IP available on the ingress node yet")
+		}
 
-		// In Minikube, we don't have the ingress gateway. Instead we do a little bit of trickery to to get the Node
-		// port.
-		n := cfg.Istio.Settings().IngressNamespace
-		if env.Settings().Minikube {
-			pods, err := env.GetPods(n, fmt.Sprintf("istio=%s", istioLabel))
-			if err != nil {
-				return nil, false, err
-			}
+		svc, err := env.Accessor.GetService(n, serviceName)
+		if err != nil {
+			return nil, false, err
+		}
 
-			scopes.Framework.Debugf("Querying ingress, pods:\n%v\n", pods)
-			if len(pods) == 0 {
-				return nil, false, errors.New("no ingress pod found")
-			}
+		scopes.Framework.Debugf("Found service for the gateway:\n%v\n", svc)
+		if len(svc.Spec.Ports) == 0 {
+			return nil, false, fmt.Errorf("no ports found in service: %s/%s", n, "istio-ingressgateway")
+		}
 
-			scopes.Framework.Debugf("Found pod: \n%v\n", pods[0])
-			ip := pods[0].Status.HostIP
-			if ip == "" {
-				return nil, false, errors.New("no Host IP available on the ingress node yet")
-			}
+		port := svc.Spec.Ports[0].NodePort
 
-			svc, err := env.Accessor.GetService(n, serviceName)
-			if err != nil {
-				return nil, false, err
-			}
+		return fmt.Sprintf("http://%s:%d", ip, port), true, nil
+	}
 
-			scopes.Framework.Debugf("Found service for the gateway:\n%v\n", svc)
-			if len(svc.Spec.Ports) == 0 {
-				return nil, false, fmt.Errorf("no ports found in service: %s/%s", n, "istio-ingressgateway")
-			}
+	// Otherwise, get the load balancer IP.
+	svc, err := env.Accessor.GetService(n, serviceName)
+	if err != nil {
+		return nil, false, err
+	}
 
-			port := svc.Spec.Ports[0].NodePort
+	if len(svc.Status.LoadBalancer.Ingress) == 0 || svc.Status.LoadBalancer.Ingress[0].IP == "" {
+		return nil, false, fmt.Errorf("service ingress is not available yet: %s/%s", svc.Namespace, svc.Name)
+	}
 
-			return fmt.Sprintf("http://%s:%d", ip, port), true, nil
-		}
+	ip := svc.Status.LoadBalancer.Ingress[0].IP
+	return fmt.Sprintf("http://%s", ip), true, nil
+}
 
-		// Otherwise, get the load balancer IP.
-		svc, err := env.Accessor.GetService(n, serviceName)
-		if err != nil {
-			return nil, false, err
-		}
+// getHTTPSAddress returns the ingress gateway address for https requests.
+func getHTTPSAddress(env *kube.Environment, cfg Config) (interface{}, bool, error) {
```

Comment:
If HTTPS is not supported on MiniKube, please write it down in the comment, and return error in this function?

Reply:
All the tests have t.Skip, if the test is running in MiniKube the test will skip and pass. The output shows it is skipped.
I added an error return in this method in case people use this method in their ingress HTTPS test.